testutil/kubernetes_helper.go

```diff
@@ -164,21 +165,26 @@ func (h *KubernetesHelper) CheckService(namespace string, serviceName string) er
 	return err
 }
 
-// GetJobStatus gets the status of a job running in a namespace. If the job does
-// not exist it return an error.
-func (h *KubernetesHelper) GetJobStatus(namespace, jobName string) (string, error) {
-	job, err := h.clientset.BatchV1().Jobs(namespace).Get(jobName, metav1.GetOptions{})
+// GetPodsForDeployment returns all pods for the given deployment
```

Comment:
could/should this be implemented in terms of `k8s.api.GetPodsFor()` ?

Reply:
Good question -- my preference is to use the synchronous APIs in tests, without any caching. I'd also like for the testutil code to be mostly independent of the rest of our runtime code, so that we avoid situations where we write a bug in some piece of code but don't detect it because we're using the same piece of code to validate the outcome in integration tests.